apps/dashboard/src/components/workflow-editor/steps/digest/digest-window.tsx

```diff
@@ -0,0 +1,96 @@
+export const DigestWindow = () => {
+  const { step } = useStep();
+  const { saveForm } = useSaveForm();
+  const { dataSchema, uiSchema } = step?.controls ?? {};
+
+  const unitOptions = useMemo(
+    () => ((dataSchema?.anyOf?.[0] as JSONSchemaDto).properties?.[unitKey] as any).enum ?? defaultUnitValues,
+    [dataSchema]
+  );
+
+  const defaultUnitOption = useMemo(
+    () => (uiSchema?.properties?.[unitKey] as any).placeholder ?? TimeUnitEnum.SECONDS,
+    [uiSchema?.properties]
+  );
+
+  return (
+    <div className="flex flex-col gap-2">
+      <FormLabel>
+        <span className="flex items-center gap-1">
+          <RiCalendarScheduleFill className="size-4" />
+          <span>Digest window</span>
+        </span>
+      </FormLabel>
+      <Tabs defaultValue="editor" className="flex h-full flex-1 flex-col">
+        <div className="bg-neutral-alpha-50 flex flex-col rounded-lg border border-solid border-neutral-100">
+          <div className="rounded-t-lg p-2">
+            <TabsList className="w-full">
+              <Tooltip>
+                <TooltipTrigger className="ml-1" asChild>
+                  <span className="flex-1">
+                    <TabsTrigger value="editor" className="w-full text-xs">
+                      Fixed duration
+                    </TabsTrigger>
+                  </span>
+                </TooltipTrigger>
+                <TooltipContent className="max-w-48" side="top" sideOffset={10}>
+                  <span>
+                    Digest begins after the last sent digest, collecting events until the set time, then sends a
+                    summary.
+                  </span>
+                </TooltipContent>
+              </Tooltip>
+              <Tooltip>
+                <TooltipTrigger className="ml-1" asChild>
+                  <span className="flex-1">
+                    <TabsTrigger value="preview" className="w-full text-xs">
+                      Interval
+                    </TabsTrigger>
+                  </span>
+                </TooltipTrigger>
+                <TooltipContent className="max-w-48" side="top" sideOffset={10}>
+                  <span>Digest aggregates events within the selected time period.</span>
+                </TooltipContent>
+              </Tooltip>
+            </TabsList>
+          </div>
+          <Separator className="bg-neutral-100" />
+          <div className="bg-background rounded-b-lg p-2">
+            <TabsContent value="editor">
+              <div className="flex items-center justify-between">
+                <span className="text-foreground-600 text-xs font-medium">Digest events for</span>
+                <AmountInput
+                  fields={{ inputKey: `controlValues.${amountKey}`, selectKey: `controlValues.${unitKey}` }}
+                  options={unitOptions}
+                  defaultOption={defaultUnitOption}
+                  className="[&_input]:!w-[3ch] [&_input]:!min-w-[3ch]"
+                  onValueChange={saveForm}
```

Comment:
The error message seems incorrectly positioned.

![image](https://github.com/user-attachments/assets/2fe18161-89b9-4c76-9163-00acb1133e50)

Reply:
`<DelayNode/>` is missing `<NodeError/>`

![image](https://github.com/user-attachments/assets/531aabdd-0d72-4481-ba8a-799520fbddb6)

Reply:
thanks, looking into it! 🙌